google-oauth-client/src/main/java/com/google/api/client/auth/openidconnect/IdTokenVerifier.java

```diff
@@ -534,7 +560,7 @@ public Map<String, PublicKey> load(String certificateUrl) throws Exception {
             Level.WARNING,
             "Failed to get a certificate from certificate location " + certificateUrl,
             io);
-        return ImmutableMap.of();
+        throw io;
```

Comment:
Memo for myself.

Now this throws an exception. With Guava's LoadingCache, the exception surfaces as ExecutionException in `publicKeyCache.get` in `verifySignature` method. This `verifySignature` method wraps it as `VerificationException`. The `verify` method catches `VerificationException`, logs the SEVERE-level message, and returns false.

With this change the LoadingCache now stops caching failure result. A potential problem would be the library users calling the verify method repeatedly in a situation where the certificate location is inaccessible. From the caller's perspective, the result (getting `false`) remains the same. In that case `IdTokenVerifier$PublicKeyLoader` logs the warning message. The user can easily tell the problem. This should be fine.

Reply:
Yep, had similar thoughts on this